// keycloak-angular/esm2015/lib/core/services/keycloak-auth-guard.js
import * as tslib_1 from "tslib";
export class KeycloakAuthGuard {
    constructor(router, keycloakAngular) {
        this.router = router;
        this.keycloakAngular = keycloakAngular;
    }
    canActivate(route, state) {
        return new Promise(((resolve, reject) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                this.authenticated = yield this.keycloakAngular.isLoggedIn();
                this.roles = yield this.keycloakAngular.getUserRoles(true);
                const result = yield this.isAccessAllowed(route, state);
                resolve(result);
            }
            catch (error) {
                reject('An error happened during access validation. Details:' + error);
            }
        })));
    }
}
if (false) {
    KeycloakAuthGuard.prototype.authenticated;
    KeycloakAuthGuard.prototype.roles;
    KeycloakAuthGuard.prototype.router;
    KeycloakAuthGuard.prototype.keycloakAngular;
    KeycloakAuthGuard.prototype.isAccessAllowed = function (route, state) { };
}
